providers/keycloak.go

```diff
@@ -95,5 +128,29 @@ func (p *KeycloakProvider) GetEmailAddress(ctx context.Context, s *sessions.Sess
 		}
 	}
 
+	if len(p.Roles) > 0 {
+		claims := make(map[string]interface{})
+
+		// Decode JWT token without verifying the signature
+		token, err := jwt.ParseSigned(s.AccessToken)
+
+		if err != nil {
+			logger.Printf("failed to parse token %s", err)
+			return "", nil
+		}
+
+		// Parse claims
+		if err := token.UnsafeClaimsWithoutVerification(&claims); err != nil {
+			logger.Printf("failed to parse claims %s", err)
+		}
+
+		var roles = ExtractRolesFromClaims(claims)
+
+		if !isSubarray(roles, p.Roles) {
```

Comment:
Nit
```suggestion
		roles := ExtractRolesFromClaims(claims)
		if !isSubarray(roles, p.Roles) {
```

Reply:
Thank you for the suggestion, I changed it :)